src/contracts/examples/redstone-oev/RedstoneAdapterAtlasWrapper.sol

```diff
@@ -88,67 +100,68 @@ contract RedstoneAdapterAtlasWrapper is Ownable, MergedSinglePriceFeedAdapterWit
         uint256 minIntervalBetweenUpdates = getMinIntervalBetweenUpdates();
         if (currentBlockTimestamp < blockTimestampFromLatestUpdate + minIntervalBetweenUpdates) {
             revert MinIntervalBetweenUpdatesHasNotPassedYet(
-                currentBlockTimestamp,
-                blockTimestampFromLatestUpdate,
-                minIntervalBetweenUpdates
+                currentBlockTimestamp, blockTimestampFromLatestUpdate, minIntervalBetweenUpdates
             );
         }
     }
 
-    function latestAnswer() public view virtual override returns (int256) {
+    function latestAnswerAndTimestamp() internal view virtual returns (int256, uint256) {
         bytes32 dataFeedId = getDataFeedId();
-        uint256 baseAnswer = IAdapter(BASE_ADAPTER).getValueForDataFeed(dataFeedId);
+        (, uint256 atlasLatestBlockTimestamp) = getTimestampsFromLatestUpdate();
         uint256 atlasAnswer = getValueForDataFeed(dataFeedId);
-        if (atlasAnswer == 0) {
-            return int256(baseAnswer);
+
+        if (atlasLatestBlockTimestamp >= block.timestamp - BASE_FEED_DELAY) {
+            return (int256(atlasAnswer), atlasLatestBlockTimestamp);
         }
-        (uint256 baseLatestTimestamp,) = IAdapter(BASE_ADAPTER).getTimestampsFromLatestUpdate();
-        (uint256 atlasLatestTimestamp,) = getTimestampsFromLatestUpdate();
 
-        if (atlasLatestTimestamp > baseLatestTimestamp - BASE_FEED_DELAY) {
-            return int256(atlasAnswer);
+        // return the latest base value which was recorded before or = block.timestamp - BASE_FEED_DELAY
+        uint80 roundId = IFeed(BASE_FEED).latestRound();
```

Comment:
I think this could be ```uint80 roundId = IFeed(BASE_FEED).latestRound() - 1;``` since weve already checked the latest round?

Reply:
if the latest round was before `block.timestamp - BASE_FEED_DELAY`, then we should use the latest round answer.